### PR TITLE
Add docs for path-regex annotation for mergeable ingresses

### PR DESCRIPTION
**Pull request by jjngx**
### Proposed changes

This PR updates docs and add examples for `path-regex` annotation feature for Mergeable Ingress.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: examples/ingress-resources/path-regex/README.md [hunk 1]
```diff
@@ -41,7 +41,9 @@ as a case-insensitive regular expression match
 - **=** : Equal sign modifier (annotation value ``exact``) - the location is considered a match if the request
  URI exactly matches the location provided.
 
-## Example 1: Case Sensitive RegEx
+## Path-Regex Annotation in Ingress Types
```
Comment (vepatel) [inline examples/ingress-resources/path-regex/README.md:44]:
```suggestion
## Path-Regex Annotation in Simple Ingress Type
```

File: docs/content/configuration/ingress-resources/advanced-configuration-with-annotations.md [hunk 2]
```diff
@@ -112,7 +112,7 @@ The table below summarizes the available annotations.
...
-|``nginx.org/path-regex`` | N/A | Enables regular expression modifiers for [location](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) directive. You can specify one of these values: "case_sensitive", "case_insensitive" or "exact". The annotation is applied to the entire Ingress resource and its paths. |  |  [Path Regex](https://github.com/nginxinc/kubernetes-ingress/tree/examples/ingress-resources/path-regex). |
+|``nginx.org/path-regex`` | N/A | Enables regular expression modifiers for [location](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) directive. You can specify one of these values: "case_sensitive", "case_insensitive", or "exact". The annotation is applied to the entire Ingress resource and its paths. If you use a Master and Minion Ingresses (Mergeable Ingresses) you can specify different regex annotations values on all minions or a subset of minions and/or a master. |  |  [Path Regex](https://github.com/nginxinc/kubernetes-ingress/tree/examples/ingress-resources/path-regex). |
```
Comment (vepatel) [inline docs/content/configuration/ingress-resources/advanced-configuration-with-annotations.md:115]:
```suggestion
|``nginx.org/path-regex`` | N/A | Enables regular expression modifiers for [location](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) directive. You can specify one of these values: "case_sensitive", "case_insensitive", or "exact". The annotation is applied to the entire Ingress resource and its paths. While using Master and Minion Ingresses i.e. Mergeable Ingresses, this annotation can be specified on both types at once but value from Minion takes precedence. |  |  [Path Regex](https://github.com/nginxinc/kubernetes-ingress/tree/examples/ingress-resources/path-regex). |
```

File: examples/ingress-resources/path-regex/README.md [hunk 3]
```diff
@@ -189,3 +191,305 @@ Corresponding NGINX config file snippet:
...
+
+Content of `coffee-minion.yaml`:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: cafe-ingress-coffee-minion
+  annotations:
+    nginx.org/mergeable-ingress-type: "minion"
```
Comment (brianehlert) [inline examples/ingress-resources/path-regex/README.md:232]: I see that path-regex is set at the master but not at the minion.
Therefore I would not expect the minion paths to support regex, only paths defined in the master.
I would expect this path to cause an error.

Comment (brianehlert) [reply examples/ingress-resources/path-regex/README.md:232]: Each ingress is a configuration to itself.  They just get flattened into a single larger server block configuration.

Comment (vepatel) [reply examples/ingress-resources/path-regex/README.md:232]: @brianehlert we don't support that particular use-case by design, where one have have a master with its own paths on top of minions so imo this behaviour of inheriting master annotation makes sense.
If one tries to add paths to master:
`spec.rules[0].http.paths: Too many: 1: must have at most 0 items`
`Ingress master is invalid or doesn't exist`

File: examples/ingress-resources/path-regex/README.md [hunk 4]
```diff
@@ -189,3 +191,305 @@ Corresponding NGINX config file snippet:
...
+spec:
+  ingressClassName: nginx
+  tls:
+  - hosts:
+    - cafe.example.com
+    secretName: cafe-secret
+  rules:
+  - host: cafe.example.com
+```
+
```
[the remark this answers is not in the document]
Comment (vepatel) [reply examples/ingress-resources/path-regex/README.md:223]: same as above

**Review by vepatel — APPROVED** [review]

**Review by shaun-nx — APPROVED** [review]

File: examples/ingress-resources/path-regex/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jjngx) [reply examples/ingress-resources/path-regex/README.md:223]: updated

File: examples/ingress-resources/path-regex/README.md (hunk 3 above)
Comment (jjngx) [reply examples/ingress-resources/path-regex/README.md:232]: fixed

**Pull request closed without merging**